charts/generic-prometheus-alerts/ci/tests/ingress-alerts-tests.yaml

```diff
@@ -0,0 +1,170 @@
+---
+rule_files:
+  - ../ingress-alerts.yaml
+
+evaluation_interval: 1m
+
+tests:
+  - interval: 1m
+    input_series:
+      - series: 'nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status="500"}'
+        values: '0 1 3'
+      - series: 'nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status="501"}'
+        values: '0 2 5'
+
+    name: 5xxErrorResponses alert fired as request rate increases over the duration of the 1 minute time range window
+
+    promql_expr_test:
+      # assert that
+      #   at 2m the rate (increase per second) in the last minute: for 500 requests it's 0.03 ((3 - 1) / 60), for 501 requests it's 0.05 ((5 - 2) / 60)
+      #   which multiplied by 60 is greater than 0
+      #   the average of those is 2.5
+      - expr: avg(rate(nginx_ingress_controller_requests{exported_namespace="test-application-dev", ingress="test-application-ingress", status=~"5.*"}[1m]) * 60 > 0) by (ingress, exported_namespace)
+        eval_time: 2m
+        exp_samples:
+          - labels: '{exported_namespace="test-application-dev", ingress="test-application-ingress"}'
+            value: 2.5
```

Comment:
This is great! 😄 

I wonder if it's possible to use the expression in the alert instead of duplicating to prevent drift? (though admittedly you'd lose the readability you have here)

Reply:
That's a great suggestion! Unfortunately it doesn't seem to be possible as `promql_expr_test.expr` needs to be a valid PromQL query, rather than being able to reference a rule's `expr` 😞 

Like you said drift may happen - if someone changes a rule's query then these `promql_expr_test` would no longer match, but then `alert_rule_test` would likely fail and we still have some regression test coverage?